SKYKitChat/Classes/UI/ConversationView/SKYChatConversationViewController.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  SKYChatConversationViewController.swift
+//  SKYKitChat
+//
+//  Copyright 2016 Oursky Ltd.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import JSQMessagesViewController
+
+class SKYChatConversationViewController: JSQMessagesViewController {
+    public var skygear: SKYContainer = SKYContainer.default()
+    public var conversation: SKYConversation?
+    public var participants: [String: SKYRecord] = [:]
+    public var messages: [SKYMessage] = []
+
+    static let errorDomain: String = "SKYChatConversationViewControllerErrorDomain"
+    var errorCreator: SKYErrorCreator {
+        return SKYErrorCreator(defaultErrorDomain: SKYChatConversationViewController.errorDomain)
+    }
+}
+
+// MARK: - Initializing
+
+extension SKYChatConversationViewController {
+
+    public class func create() -> SKYChatConversationViewController {
+        return SKYChatConversationViewController(nibName: "JSQMessagesViewController",
+                                                 bundle: Bundle(for: JSQMessagesViewController.self))
+    }
+}
+
+// MARK: - Lifecycle
+
+extension SKYChatConversationViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        SKYChatConversationViewController.nib().instantiate(withOwner: self, options: nil)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        guard self.conversation != nil else {
+            print("Conversation is not set")
+            self.dismiss(animated: animated)
+            return
+        }
+
+        self.fetchParticipants(completion: nil)
+        self.fetchMessages(before: nil, completion: nil)
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+
+    }
+
+    func dismiss(animated: Bool) {
+        if let nc = self.navigationController, let topVC = nc.topViewController {
+            guard self == topVC else {
+                // I am not the top view controller
+                return
+            }
+
+            nc.popViewController(animated: animated)
+        } else {
+            self.dismiss(animated: animated, completion: nil)
+        }
+    }
+}
+
+// MARK: - Utility Methods
+
+extension SKYChatConversationViewController {
+
+    open func fetchParticipants(completion: (([SKYRecord]?, Error?) -> Void)?) {
```

Comment:
Why is the completion optional? This method and `fetchMessages` are useless when completion is nil